GameOfLife.java

```diff
@@ -59,20 +60,40 @@ public static void play(String fileName) {
 	// of this frame as representing the infinite number of dead cells that exist in every direction.
 	// This function assumes that the input file contains valid data, and does no input testing.
 	public static int[][] read(String fileName) {
-		In in = new In(fileName); // Constructs an In object for reading the input file
+		In in = new In(fileName); // Constructs an In object for reading 
 		int rows = Integer.parseInt(in.readLine());
 		int cols = Integer.parseInt(in.readLine());
 		int[][] board = new int[rows + 2][cols + 2];
-		//// Replace the following statement with your code.
-		return null;
+		String textRow;
+		int currow = 1, curcol;
+
+		while(!in.isEmpty()){
+			textRow = in.readLine();
+			curcol = 1;
+			if (textRow != "") {
+				for(int i=0; i<textRow.length(); i++){
```

Comment:
don't need this if. If textRow == "" then textRow.length() will be 0 and the loop will not run